packages/sentinel_one/data_stream/activity/elasticsearch/ingest_pipeline/default.yml

```
@@ -28,6 +22,58 @@ processors:
         - json.id 
       target_field: _id
       ignore_missing: true
+  - script:
+      description: Set the value of event.category, event.outcome and event.type.
+      lang: painless
+      source: >
+          def eventCategory = new HashSet();
+          def eventType = new HashSet();
+          if(ctx.json?.threatId != null && ctx.json?.threatId != "") {
+              if(ctx.json?.data?.threatClassification == "Malware") {
+                eventCategory.add("malware");
+                eventType.add("info");
+             }
+             else{
+                 eventCategory.add("threat");
+                 eventType.add("indicator");
+             }
+          }
+          else if(ctx.json?.primaryDescription != null){
+              if(ctx.json?.primaryDescription?.toLowerCase().contains("logged in")){
```

This could be made less verbose and made to do less work by assigning a local. Also, since we know `ctx.json?.primaryDescription != null`, the conditional null derefs are not needed in the later steps.

```suggestion
          else if(ctx.json?.primaryDescription != null) {
              def description = ctx.json.primaryDescription.toLowerCase();
              if (description.contains("logged in")) {
```
with similar uses in all the following conditions.